formats/dot/internal/errors/errors.go

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// generated by gocc; DO NOT EDIT.
+// Code generated by gocc; DO NOT EDIT.
```

Comment:
I think the copyright header does need to be on generated code. The generated code is the artifact, the process to generate it may be punching keys, or getting a machine to pretend to punch keys. The path to the product is not really relevant.

The way that ragel works is that the generated code is produced from a template ragel file. An example ragel file [here](https://github.com/cayleygraph/cayley/blob/master/quad/nquads/raw.rl) which generates [this](https://github.com/cayleygraph/cayley/blob/master/quad/nquads/raw.go). Although even that inserts a line directive - in a way that actually explicitly does not match the canonical form.

I guess the best approach is to add some code to the make scripts that will replace the warning with the warning suffixed with the copyright notice.

Sorry for all this rigmarole.